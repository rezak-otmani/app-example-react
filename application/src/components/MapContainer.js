import React, { Component } from 'react';
import {Map, Marker, GoogleApiWrapper} from 'google-maps-react';


export class MapContainer extends Component {


render (){

    return (
      <Map 
           
          google={this.props.google}       
          style={{width: '100vw', height: '100vh', position: 'relative'}}
           initialCenter={{
            lat: 36.73556051,
            lng: 3.95799242
          }} 
          zoom={17}>
           
          
         <Marker
 	   title={'DATO'}
   	   position={{lat: 36.73556051, lng: 3.95799242}}
  	   name={'DATO'}
       	/>
             
       
         
      </Map>
    );
  }
}

export default GoogleApiWrapper({
  apiKey: 'AIzaSyAyesbQMyKVVbBgKVi2g6VX7mop2z96jBo'
})(MapContainer)
