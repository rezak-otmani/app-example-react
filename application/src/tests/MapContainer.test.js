import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MapContainer from '../components/MapContainer';


describe('Google Map Component', () => {

  let wrapper;
 
  beforeEach(() => {
    wrapper = shallow(
      <MapContainer />
    );
  });

          it('it should render the Map ', () => {
                  expect(wrapper.length).toBe(1);
         
           });

          it('it should have a Map component', () => {
                const map = wrapper.find('Map').first();
                 expect((map).length).toBe(1);
               
	  });
        
          it('it should have a Marker component', () => {
                 const marker = wrapper.find('Marker').first();
                 expect((marker).length).toBe(1);

           });


              
            it('renders correctly', () => {
                const tree = renderer.create(
                 <MapContainer />
                          ).toJSON();
                  expect(tree).toMatchSnapshot();
             });



});
