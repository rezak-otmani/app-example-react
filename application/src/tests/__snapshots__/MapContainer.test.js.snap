// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <div
    className={undefined}
    style={
      Object {
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "bottom": 0,
          "display": "inherit",
          "height": "100vh",
          "left": 0,
          "position": "relative",
          "right": 0,
          "top": 0,
          "width": "100vw",
        }
      }
    >
      Loading map...
    </div>
  </div>
  <div />
</div>
`;
